### PR TITLE
[IOTDB-5671] add SPRINTZ and RLBE encodor and LZMA2 compressor

### PR DESCRIPTION
**Pull request by xjz17**
add SPRINTZ and RLBE encodor and LZMA2 compressor

* SPRINTZ

The SPRINTZ encoding combines encodings in four steps: predicting, bit-packing, run-length encoding and entropy encoding. SPRINTZ algorithm is suitable for predictable time series. For delta function, the vast repeats or linearly increasing time series is the best target.

* RLBE

The RLBE encoding proposes to combine delta, run-length and Fibonacci based encoding ideas. It has five steps: differential coding, binary encoding, run-length, Fibonacci coding and concatenation.
It is more suitable for the differential value of time series is positive and small.

* LZMA2

LZMA2  is a compression algorithm that is widely used for compressing files and data. It is an extension of the original LZMA algorithm, which was developed by Igor Pavlov in 1998. The LZMA2 algorithm was introduced as part of the 7-Zip archiver software and is now supported by many other compression tools.

LZMA2 is a powerful compression algorithm that is well-suited for applications where high compression ratios are important and speed is not a primary concern. It is commonly used in file archiving, backup and restore operations, and other data compression applications.

### CODE REVIEW

File: tsfile/pom.xml [hunk 1]
```diff
@@ -104,6 +110,14 @@
...
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.tukaani</groupId>
+            <artifactId>xz</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.tukaani</groupId>
+            <artifactId>xz</artifactId>
+        </dependency>
```
Comment (HTHou) [inline tsfile/pom.xml:120]: Duplicated?

File: pom.xml [hunk 2]
```diff
@@ -517,6 +517,12 @@
                 <artifactId>zstd-jni</artifactId>
                 <version>${zstd-jni.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.tukaani</groupId>
+                <artifactId>xz</artifactId>
+                <version>1.9</version>
+                <scope>compile</scope>
+            </dependency>
```
Comment (HTHou) [inline pom.xml:525]: Indicate the version number like `${xz.version}` ?

Comment (xjz17) [reply pom.xml]: ok, revised it

File: tsfile/pom.xml (hunk 1 above)
Comment (xjz17) [reply tsfile/pom.xml]: ok, deleted duplicate one

File: tsfile/pom.xml [hunk 3]
```diff
@@ -77,6 +77,12 @@
             <groupId>io.airlift</groupId>
             <artifactId>slice</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.tukaani</groupId>
+            <artifactId>xz</artifactId>
+            <version>1.9</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (HTHou) [inline tsfile/pom.xml:85]: It also repeats the line 114. Consider removing it?

Comment (xjz17) [reply tsfile/pom.xml:85]: OK， removed it

**Comment by HTHou** [comment]
And don't forget to run `mvn spotless:apply` to fix the code format issue.

File: tsfile/pom.xml (hunk 3 above)
Comment (HTHou) [inline tsfile/pom.xml:85]:
```suggestion
        <dependency>
            <groupId>org.tukaani</groupId>
            <artifactId>xz</artifactId>
            <scope>compile</scope>
        </dependency>
```

File: docs/UserGuide/Reference/Common-Config-Manual.md [hunk 4]
```diff
@@ -1212,12 +1212,12 @@ Different configuration parameters take effect in the following three ways:
...
-|    Name     | compressor                                            |
-|:-----------:|:------------------------------------------------------|
-| Description | Data compression method                               |
-|    Type     | Enum String : “UNCOMPRESSED”, “SNAPPY”, "LZ4", "ZSTD" |
-|   Default   | SNAPPY                                                |
-|  Effective  | hot-load                                              |
+|    Name     | compressor                                                     |
+|:-----------:|:---------------------------------------------------------------|
+| Description | Data compression method                                        |
+|    Type     | Enum String : “UNCOMPRESSED”, “SNAPPY”, "LZ4", "ZSTD", "lZMA2" |
```
Comment (HTHou) [inline docs/UserGuide/Reference/Common-Config-Manual.md:1218]:
```suggestion
|    Type     | Enum String : "UNCOMPRESSED" "SNAPPY", "LZ4", "ZSTD", "LZMA2" |
```

File: docs/zh/UserGuide/Reference/Common-Config-Manual.md [hunk 5]
```diff
@@ -1251,12 +1251,12 @@ IoTDB ConfigNode 和 DataNode 的公共配置参数位于 `conf` 目录下。
...
-|   名字   | compressor                                          |
-|:------:|:----------------------------------------------------|
-|   描述   | 数据压缩方法                                              |
-|   类型   | 枚举 String : “UNCOMPRESSED”, “SNAPPY”, “LZ4”, “ZSTD” |
-|  默认值   | SNAPPY                                              |
-| 改后生效方式 | 热加载                                                 |
+|   名字   | compressor                                                   |
+|:------:|:-------------------------------------------------------------|
+|   描述   | 数据压缩方法                                                       |
+|   类型   | 枚举 String : “UNCOMPRESSED”, “SNAPPY”, “LZ4”, “ZSTD”, "LZMA2" |
```
Comment (HTHou) [inline docs/zh/UserGuide/Reference/Common-Config-Manual.md:1257]:
```suggestion
|   类型   | 枚举 String : "UNCOMPRESSED", "SNAPPY", "LZ4", "ZSTD", "LZMA2" |
```

File: tsfile/src/main/java/org/apache/iotdb/tsfile/compress/IUnCompressor.java
[the remark this answers is not in the document]
Comment (xjz17) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/compress/IUnCompressor.java:1]: added it

**Comment by shuwenwei** [comment]
add LICENSE-binary
https://github.com/apache/iotdb/blob/master/LICENSE-binary

**Review by HTHou — APPROVED** [review]

**Review by shuwenwei — APPROVED** [review]

**Pull request merged**